docs/conf.py

```diff
@@ -238,7 +241,91 @@
                       'issue '),
             'fileref': ('https://github.com/NVIDIA/DALI/tree/' + (git_sha if git_sha != u'0000000' else "master") + '/%s', ''),}
 
+
+from typing import (
+    Any, Callable, Dict, Iterator, List, Optional, Sequence, Set, Tuple, Type, TypeVar, Union
+)
+from typing import get_type_hints
+
+
+_dali_enums = ["DALIDataType", "DALIIterpType", "DALIImageType", "PipelineAPIType"]
+
+class EnumDocumenter(ClassDocumenter):
+    # Register as .. autoenum::
+    objtype = 'enum'
+    # Produce .. py:class:: fields in the RST doc
+    directivetype = 'class'
+    # priority = 10
+
+    def __init__(self, *args: Any) -> None:
+        super().__init__(*args)
+
+    @classmethod
+    def can_document_member(cls, member: Any, membername: str, isattr: bool, parent: Any
+                            ) -> bool:
+        """Verify that we handle only the registered DALI enums. Pybind doesn't subclass Enum class,
+        so we need an explicit list.
+        """
+        return membername in _dali_enums and isinstance(parent, ClassDocumenter)
```

Comment:
Do we need to check `isinstance(parent, ClassDocumenter)` ?
As `EnumDocumenter` inherits from `ClassDocumenter`, can parent be different?

Reply:
I'm not sure, but at some point a lot of elements were checked against this class, I can check if it's really necessary, but maybe it's safer to just keep it.

Reply:
Sure.